File: server/datasets/python_source-code/Multiple-C1/renamed_1.py
"""
Graph BFS - Renamed Submission
Same logic as organic_2.py, but every identifier has been renamed:
build_graph -> construct_adjacency_map, breadth_first_search -> traverse_bfs,
graph -> adjacency_map, seen -> discovered, queue -> pending_nodes, etc.
"""

from collections import defaultdict


def construct_adjacency_map(edge_pairs):
    adjacency_map = defaultdict(list)
    for source_node, dest_node in edge_pairs:
        adjacency_map[source_node].append(dest_node)
        adjacency_map[dest_node].append(source_node)
    return adjacency_map


def traverse_bfs(adjacency_map, origin):
    discovered = {origin}
    pending_nodes = [origin]
    visitation_order = []

    while pending_nodes:
        active_node = pending_nodes.pop(0)
        visitation_order.append(active_node)

        for linked_node in adjacency_map[active_node]:
            if linked_node not in discovered:
                discovered.add(linked_node)
                pending_nodes.append(linked_node)

    return visitation_order


if __name__ == "__main__":
    edge_pairs = [("A", "B"), ("A", "C"), ("B", "D"), ("B", "E"), ("C", "F"), ("E", "F")]
    adjacency_map = construct_adjacency_map(edge_pairs)
    print(traverse_bfs(adjacency_map, "A"))

File: server/datasets/java_source-code/Multiple-C1/renamed_1.java
/*
 * Graph BFS - Renamed Submission
 * Same logic as organic_2.java, but every identifier has been renamed:
 * buildGraph -> constructAdjacencyMap, breadthFirstSearch -> traverseBfs,
 * seen -> discovered, queue -> pendingNodes, traversal -> visitationOrder.
 */

import java.util.*;

class Solution {
    public static Map<String, List<String>> constructAdjacencyMap(String[][] edgePairs) {
        Map<String, List<String>> adjacencyMap = new HashMap<>();
        for (String[] pair : edgePairs) {
            String sourceNode = pair[0];
            String destNode = pair[1];
            adjacencyMap.computeIfAbsent(sourceNode, k -> new ArrayList<>()).add(destNode);
            adjacencyMap.computeIfAbsent(destNode, k -> new ArrayList<>()).add(sourceNode);
        }
        return adjacencyMap;
    }

    public static List<String> traverseBfs(Map<String, List<String>> adjacencyMap, String origin) {
        Set<String> discovered = new HashSet<>();
        List<String> pendingNodes = new ArrayList<>();
        List<String> visitationOrder = new ArrayList<>();

        discovered.add(origin);
        pendingNodes.add(origin);

        while (!pendingNodes.isEmpty()) {
            String activeNode = pendingNodes.remove(0);
            visitationOrder.add(activeNode);

            List<String> links = adjacencyMap.getOrDefault(activeNode, Collections.emptyList());
            for (String linkedNode : links) {
                if (!discovered.contains(linkedNode)) {
                    discovered.add(linkedNode);
                    pendingNodes.add(linkedNode);
                }
            }
        }
        return visitationOrder;
    }

    public static void main(String[] args) {
        String[][] edgePairs = {
            {"A", "B"}, {"A", "C"}, {"B", "D"}, {"B", "E"}, {"C", "F"}, {"E", "F"}
        };
        Map<String, List<String>> adjacencyMap = constructAdjacencyMap(edgePairs);
        System.out.println(traverseBfs(adjacencyMap, "A"));
    }
}
